lib/ngx/re.lua

```diff
@@ -84,6 +84,9 @@ end
 
 
 function _M.split(subj, regex, opts, ctx, max, res)
+    if subj == nil then
+        return nil, nil
```

Comment:
I don't think I like it. If you really want to detect `nil` input, you should really throw an error here instead instead of passing along `nil`.